### ISSUE TITLE
Multi threading support or use in parallel with tinyrecord

### ISSUE THREAD

**Issue by dan-cristian** [schapman1974/tinymongo#53]
This is a question rather than an issue, as I could not find an answer.

Does tinymongo support multithreading and if not, can I use it in conjunction with [tinyrecord ](https://github.com/eugene-eeo/tinyrecord/) module?

I would like to store values in database performing inserts and updates from threads while being able to use [Flask-Admin](https://github.com/flask-admin/flask-admin/tree/master/examples/tinymongo) interface to perform UI CRUD operations.

**Comment by dan-cristian** [comment]
Run some tests and it seems that tinymongo does not support multithreading. However tinyrecord can be used to add multithreading support but code changes are needed in class TinyMongoCollection(object).

**Comment by schapman1974** [comment]
You are correct. However I would be happy to accept pull requests to add support for this.  As well think about unit testing for it

**Comment by dan-cristian** [comment]
Ok, will do, I'm in progress of adding tinyrecord support and also an
easier/shorter way to perform objects creation and handling.

**Comment by schapman1974** [comment]
Current state: resolved for TinyMongo itself. Writes now use in-process and optional inter-process file locks plus atomic temp-file replace, and the repo has opt-in integration stress tests for concurrent writers. This does not require tinyrecord for the supported TinyMongo write path. Closing this question as addressed.
